### PR TITLE
Fix acceleration in append mode with refresh_sql specified

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Fix acceleration when `append` mode is used with `refresh_sql`

Example test run: https://github.com/spiceai/spiceai/actions/runs/12046323524

## 🔨 Related Issues

N/A

## 📄 Documentation Requirements

N/A

## 🤔 Concerns

N/A

### CODE REVIEW

File: .github/workflows/e2e_test_ci.yml [hunk 1]
```diff
@@ -516,6 +519,10 @@ jobs:
         run: |
           killall spice || true
           cat spice.log
+          if grep -iE "(failed|error)" spice.log; then
```
Comment (sgrebnov) [inline .github/workflows/e2e_test_ci.yml:565]: Initial load can be successful and main verification logic will pass but refreshes can error so we check logs to identify any errors/failures

**Review by peasee — COMMENTED** [review]
Could we make these new tests, rather than editing existing tests? e.g. E2E coverage with and without `append`, rather than just with `append`.

File: test/scripts/setup-data-mysql.sql [hunk 2]
```diff
@@ -106,4 +88,26 @@ INSERT INTO test_mysql_table (
   NULL,
   NULL,
   NULL
+),
+-- Third dynamically generated test row with realistic values
```
Comment (sgrebnov) [inline test/scripts/setup-data-mysql.sql]:
```suggestion
-- Third test row
```

File: test/scripts/setup-data-postgresql.sql [hunk 3]
```diff
@@ -72,11 +75,33 @@ INSERT INTO test_postgresql_table (
...
   NULL,
   NULL,
   NULL,
   NULL,
   NULL,
   NULL
-);
\ No newline at end of file
+),
+-- Third dynamically generated test row with realistic values
```
Comment (sgrebnov) [inline test/scripts/setup-data-postgresql.sql]:
```suggestion
-- Third test row
```

File: .github/workflows/e2e_test_ci.yml [hunk 4]
```diff
@@ -472,6 +474,43 @@ jobs:
           if [[ -n "$ENGINE" ]]; then
             echo "    acceleration:" >> spicepod.yaml
             echo "      enabled: true" >> spicepod.yaml
+            echo "      refresh_sql: SELECT * FROM test_postgresql_table WHERE int2_column is NULL or int2_column != 10" >> spicepod.yaml
```
Comment (sgrebnov) [inline .github/workflows/e2e_test_ci.yml:477]: Learned today that `int2_column != 10` will also filter out records where value is NULL (native engines behavior)

File: crates/runtime/src/accelerated_table/refresh_task.rs [hunk 5]
```diff
@@ -514,23 +514,19 @@ impl RefreshTask {
...
             .limit(0, Some(1))
     }
 
-    async fn accelerator_df(
+    fn accelerator_df(
         &self,
-        ctx: SessionContext,
-        sql_opt: Option<&str>,
+        ctx: &SessionContext,
+        _sql_opt: Option<&str>,
```
Comment (Jeadie) [inline crates/runtime/src/accelerated_table/refresh_task.rs]: Why keep `_sql_opt` as variable if unused?

File: crates/runtime/src/accelerated_table/refresh_task.rs [hunk 6]
```diff
@@ -550,10 +546,9 @@ impl RefreshTask {
 
         let existing_records = self
             .accelerator_df(
-                self.refresh_df_context(Arc::clone(&federated_provider)),
+                &self.refresh_df_context(Arc::clone(&federated_provider)),
```
Comment (Jeadie) [inline crates/runtime/src/accelerated_table/refresh_task.rs]: Doesn't this change imply:
1. There are no existing records, 
2. There is no overlap between existing and the new records
3. There is no need to `filter_records` [below](https://github.com/spiceai/spiceai/pull/3697/files#diff-90f336ae367603afbc75d3643de47bae6a7f49eadbdc061aeaac5d1d70b4433eR569). 

IIRC, The need to diff current/new records is to handle timestamp jitter in the underlying systems that may have been missed in the previous append refresh.

Comment (Jeadie) [reply crates/runtime/src/accelerated_table/refresh_task.rs]: Not super confident on this however.

File: crates/runtime/src/accelerated_table/refresh_task.rs (hunk 5 above)
Comment (sgrebnov) [reply crates/runtime/src/accelerated_table/refresh_task.rs]: Updated, thank you

File: crates/runtime/src/accelerated_table/refresh_task.rs (hunk 6 above)
Comment (Jeadie) [reply crates/runtime/src/accelerated_table/refresh_task.rs]: No, was mistaken.

**Review by Jeadie — APPROVED** [review]

**Review by ewgenius — APPROVED** [review]

**Review by phillipleblanc — APPROVED** [review]

**Review by lukekim — APPROVED** [review]

**Pull request merged**